streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -229,22 +242,24 @@ private void updateProcessorContext(final StampedRecord record, final ProcessorN
 
     /**
      * <pre>
-     *  - flush state and producer
-     *  - write checkpoint
-     *  - commit offsets
+     * - flush state and producer
+     * - if(!eos) write checkpoint
+     * - commit offsets and maybe start new transaction
      * </pre>
      */
     @Override
-    public void commit() {
+    public void commit(final boolean startNewTransaction) {
         log.trace("{} Committing", logPrefix);
         metrics.metrics.measureLatencyNs(
             time,
             new Runnable() {
                 @Override
                 public void run() {
                     flushState();
-                    stateMgr.checkpoint(recordCollectorOffsets());
-                    commitOffsets();
+                    if (!eosEnabled) {
```

Comment:
Good catch! I moved `eosEnabled` to base class and forgot to remove `exactlyOnceEnabled` var here.